source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/controls/connect/ServerTypeSelectControl.java

```diff
@@ -180,16 +302,31 @@ public ICommandExecutor getNoErrorDialogCommandExecutor() {
     }
 
     public URI getServer() {
+        if (this.serverURI == null) {
+            return URIUtils.VSTS_ROOT_URL;

```

Comment:
getServer is called by other classes to get the server the user selected. Null or empty would imply the user didn't select a server (i.e. the Next button should be disabled)